### PR TITLE
Update transaction execution info and description of relevant HTTP APIs

### PR DESCRIPTION
**Pull request by slowli**
This PR describes new transaction execution mechanics introduced in v0.6.

## See also
https://wiki.bf.local/display/EXN/Transaction+status

### CODE REVIEW

File: misc/lint.sh [hunk 1]
```diff
@@ -41,6 +41,7 @@ lint_html () {
   html5validator --root "$SITE_DIR" --show-warnings --ignore-re \
     'Illegal character in query: "\|" is not allowed' \
     '"(autocorrect|autocapitalize)" not allowed on element "input"' \
+    '"autocomplete" is only allowed when the input type is' \
```
Comment (alexander-irbis) [inline misc/lint.sh]: Здесь что-то осталось недосказано?

**Comment by dmitry-timofeev** [comment]
@slowli , @DarkEld3r I have a question though — has this change in Exonum affected the format of the return values of the corresponding APIs (e.g., `/transactions/<tx_hash>`)? If so, does the API description _or_ examples need to be updated?

**Comment by DarkEld3r** [comment]
@dmitry-timofeev As far as I can see, yes.

**Comment by dmitry-timofeev** [comment]
@DarkEld3r I’m not sure I understand :smile:

File: misc/lint.sh (hunk 1 above)
Comment (alexander-irbis) [reply misc/lint.sh]: Is there anything left unsaid?

Comment (alexander-irbis) [inline misc/lint.sh]: Is there anything left unsaid?

Comment (slowli) [reply misc/lint.sh]: There are a lot of `input` types in the relevant error message (see [this Travis log](https://travis-ci.org/exonum/exonum-doc/builds/358907689)), so I've skipped them.

**Comment by slowli** [comment]
@dmitry-timofeev @DarkEld3r I'll update the endpoint description within this PR. I'm thinking about removing examples altogether, or reducing them to something less brittle, a la

```javascript
{
  "type": "in-pool",
  "content": {
    "protocol_version": ...,
    "service_id": ..., // Identifier of the service processing transaction
    "message_id": ...,
    "body": ..., // JSON-encoded body
    "signature": ...
  }
}
```

**Comment by dmitry-timofeev** [comment]
@slowli , I'd keep at least one or two, because a proper JSON might be easier to grasp than a spec. If you also want to reduce the amount of space they take, I'd consider making them hidden by default (if that's possible with the tool we use to generate HTML pages).

File: misc/lint.sh [hunk 2]
```diff
@@ -41,6 +41,7 @@ lint_html () {
   html5validator --root "$SITE_DIR" --show-warnings --ignore-re \
     'Illegal character in query: "\|" is not allowed' \
     '"(autocorrect|autocapitalize)" not allowed on element "input"' \
```
Comment (Maria-Nosyk) [inline misc/lint.sh]: > '"(autocorrect|autocapitalize)" not allowed on element "input"' 
is not allowed

Comment (slowli) [reply misc/lint.sh]: This is a third-party error message, so I cannot change it :disappointed_relieved:

File: src/advanced/node-management.md [hunk 3]
```diff
@@ -462,6 +334,157 @@ All explorer API endpoints share the same base path, denoted
...
+    - `"success"` denotes a successfully completed transaction
+    - `"error"` denotes a transaction that has returned an error (for example,
+      because of transaction parameters not satisfying context-dependent checks)
+    - `"panic"` denotes a transaction that has raised a runtime exception
+      (for example, attempted to divide by zero)
+
+- **status.code**: integer  
+  Error code supplied by the service developer. Only present for erroneous
+  transactions. Has service-specific meaning
+- **status.description**: string=
```
Comment (dmitry-timofeev) [inline src/advanced/node-management.md]: Is the equality sign intentional?

File: src/advanced/node-management.md [hunk 4]
```diff
@@ -462,6 +334,157 @@ All explorer API endpoints share the same base path, denoted
...
+  Always equals to `committed`
+- **status**: Object  
+  [Transaction execution](../architecture/transactions.md#execute) status
+- **status.type**: `"success"` | `"error"` | `"panic"`  
+  Execution status kind:
+
+    - `"success"` denotes a successfully completed transaction
+    - `"error"` denotes a transaction that has returned an error (for example,
+      because of transaction parameters not satisfying context-dependent checks)
+    - `"panic"` denotes a transaction that has raised a runtime exception
```
Comment (dmitry-timofeev) [inline src/advanced/node-management.md]: @DarkEld3r ,

Just wondering if it is OK to use "panic" (a Rust-specific term) in the context of polyglot Exonum? The other question is whether unrecoverable errors shall be treated as recoverable.

Comment (DarkEld3r) [reply src/advanced/node-management.md]: @dmitry-timofeev
Good question. I think that we can use "panic", but not as a Rust-specific term. We can explain it as an Exonum transactions' specific term.

Comment (dmitry-timofeev) [reply src/advanced/node-management.md]: OK, thanks, and what about always-attempting-to-recover Exonum? 

On top of that, I asked some more questions on the wiki page above, we can discuss this question there too.

File: src/advanced/node-management.md (hunk 3 above)
Comment (slowli) [reply src/advanced/node-management.md]: Yes, it means that the field is optional, as described [at the beginning of the page](https://github.com/slowli/exonum-doc/blob/5a67307ddfa5c088f9b6e4b3c0bff5f3a153abac/src/advanced/node-management.md#types).

**Pull request merged**